lib/__tests__/plugins.test.mjs

```diff
@@ -360,3 +370,30 @@ describe('loading a plugin from options.cwd', () => {
 		expect(result.warnings()[0].rule).toBe('plugin/warn-about-foo');
 	});
 });
+
+describe('deprecated CJS plugins', () => {
+	function runLint(options = {}) {
+		const config = {
+			plugins: [absolutePath('./fixtures/plugin-warn-about-foo.cjs')],
+			rules: {},
+		};
+
+		return postcss()
+			.use(stylelint({ config, ...options }))
+			.process('a {}', { from: undefined });
+	}
+
+	it('shows the warning', async () => {
+		await runLint();
+
+		expect(consoleWarn).toHaveBeenCalledWith(
+			expect.stringContaining('plugin-warn-about-foo.cjs" is a CJS plugin and has been deprecated'),
+		);
+	});
+
+	it('does not show the warning when `quietDeprecationWarnings` is enabled', async () => {
+		await runLint({ quietDeprecationWarnings: true });
+
+		expect(consoleWarn).not.toHaveBeenCalled();
+	});
+});
```

Comment:
questions

Can we use a commonjs `.js` file fixture?
If it fails to load, can we add a test that checks the error?
If it succeeds, then the compatibility should be mentioned in the documentation.

Reply:
Yes, we can add a test case importing a CJS plugin with `.js`. This case should succeed like `.cjs` cases.

For "compatibility" you said, do you have any specific suggestion in the documentation?

Reply:
It's reassuring to know that the cjs plugins will remain compatible in a mixed environment.

> do you have any specific suggestion in the documentation?

Not really :)